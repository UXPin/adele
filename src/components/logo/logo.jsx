/* eslint-disable arrow-body-style */
import React from 'react';
import PropTypes from 'prop-types';
import SVGInline from 'react-svg-inline';
import logo from '../../icons/uxpin_logo.svg';

const Logo = ({ tab }) => {
  return (
    <a
      href="https://www.uxpin.com?utm_source=adele.uxpin.com&utm_medium=upartner&utm_campaign=Adele"
      target="_blank"
      rel="noopener noreferrer"
      title="UXPin logo"
      tabIndex={tab}
    >
      <SVGInline svg={logo} />
    </a>
  );
};

Logo.propTypes = {
  tab: PropTypes.number,
};

Logo.defaultProps = {
  tab: 0,
};

export { Logo as default };
