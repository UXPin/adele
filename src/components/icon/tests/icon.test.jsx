import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Icon from '../icon';
import twitter from '../../icons/twitter.svg';
import logo from '../../icons/uxpin_logo.svg';

it('renders small twitter icon', () => {
  const wrapper = mount(<Icon i={twitter} size={'s'} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders large uxpin icon', () => {
  const wrapper = mount(<Icon i={logo} size={'l'} color={'#000'} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders small uxpin icon rotated', () => {
  const wrapper = mount(<Icon i={logo} size={'s'} rotate={45} />);
  expect(wrapper).toMatchSnapshot();
});
