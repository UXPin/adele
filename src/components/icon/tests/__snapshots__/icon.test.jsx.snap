// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders large uxpin icon 1`] = `
.c0:focus {
  outline: 0;
}

.c0 > svg {
  height: 100%;
  fill: #777777;
  cursor: pointer;
}

<Icon
  active={false}
  color="#000"
  i="test-file-stub"
  in="blue"
  rotate={0}
  size="l"
>
  <Styled(SVGInline)
    color="#000"
    data-active={false}
    in="blue"
    rotate={0}
    size="l"
    svg="test-file-stub"
  >
    <SVGInline
      className="c0"
      classSuffix="-svg"
      cleanup={Array []}
      cleanupExceptions={Array []}
      color="#000"
      component="span"
      data-active={false}
      in="blue"
      rotate={0}
      size="l"
      svg="test-file-stub"
    >
      <span
        className="SVGInline c0"
        color="#000"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test-file-stub",
          }
        }
        data-active={false}
        in="blue"
        rotate={0}
        size="l"
      />
    </SVGInline>
  </Styled(SVGInline)>
</Icon>
`;

exports[`renders small twitter icon 1`] = `
.c0:focus {
  outline: 0;
}

.c0 > svg {
  width: 13px;
  height: 100%;
  fill: #777777;
  cursor: pointer;
}

<Icon
  active={false}
  color="#444"
  i="test-file-stub"
  in="blue"
  rotate={0}
  size="s"
>
  <Styled(SVGInline)
    color="#444"
    data-active={false}
    in="blue"
    rotate={0}
    size="s"
    svg="test-file-stub"
  >
    <SVGInline
      className="c0"
      classSuffix="-svg"
      cleanup={Array []}
      cleanupExceptions={Array []}
      color="#444"
      component="span"
      data-active={false}
      in="blue"
      rotate={0}
      size="s"
      svg="test-file-stub"
    >
      <span
        className="SVGInline c0"
        color="#444"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test-file-stub",
          }
        }
        data-active={false}
        in="blue"
        rotate={0}
        size="s"
      />
    </SVGInline>
  </Styled(SVGInline)>
</Icon>
`;

exports[`renders small uxpin icon rotated 1`] = `
.c0:focus {
  outline: 0;
}

.c0 > svg {
  width: 13px;
  height: 100%;
  fill: #777777;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  cursor: pointer;
}

<Icon
  active={false}
  color="#444"
  i="test-file-stub"
  in="blue"
  rotate={45}
  size="s"
>
  <Styled(SVGInline)
    color="#444"
    data-active={false}
    in="blue"
    rotate={45}
    size="s"
    svg="test-file-stub"
  >
    <SVGInline
      className="c0"
      classSuffix="-svg"
      cleanup={Array []}
      cleanupExceptions={Array []}
      color="#444"
      component="span"
      data-active={false}
      in="blue"
      rotate={45}
      size="s"
      svg="test-file-stub"
    >
      <span
        className="SVGInline c0"
        color="#444"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test-file-stub",
          }
        }
        data-active={false}
        in="blue"
        rotate={45}
        size="s"
      />
    </SVGInline>
  </Styled(SVGInline)>
</Icon>
`;
